Lib/_pyio.py

```diff
@@ -249,7 +249,7 @@ def open(file, mode="r", buffering=-1, encoding=None, errors=None,
                 pass
             else:
                 if bs > 1:
```

Comment:
Isn't this redundant if use the maximal value? This can be either
```py
if bs > buffering:
    buffering = bs
```
or
```py
buffering = max(bs, buffering)
```
(buffering is set to DEFAULT_BUFFER_SIZE above).

BTW, it looks like the C code always uses the block size, even if it is smaller than DEFAULT_BUFFER_SIZE. So the NEWS entry is not correct.

Reply:
I am gonna have a look at the C code and triple check what it does and what is should do.
I think open() used the block size when available (4k on Linux usually) or default 8k. The C buffering code used 8k by default and did not seem to rely on the block size. 
The python API sets the buffer size arguments when invoking the C classes, so it works as expected, but C code invoking the C API would not have the behavior (should it? it doesn't need to because the C buffering code operates on non-files too and doesn't have the same performance consideration as the python code, but I will have a look).

Reply:
Add a test that opens a large binary file with the default buffering, call `read1()` or `peek()`, and check the size of the result. It should be the same for both implementations.